core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +65,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
+    private AtomicBoolean rerouting = new AtomicBoolean();
+
+    private volatile long registeredNextDelaySetting = Long.MAX_VALUE;
+    private volatile ScheduledFuture registeredNextDelayFuture;
```

Comment:
this is updated in the cluster state update thread right? I think it doesn't need to be volatile then?